src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -40,10 +46,23 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 			return nil
 		}); err != nil {
 			log.Error(err, "Failed database transaction to check cluster member role")
+			txnRetries++
 		}
+
+		if txnRetries >= a.config.MaxPreRemoveHookDBTxnRetries {
+			return false, ErrMaxRemoveTxnRetries
+		}
+
 		return notPending, nil
 	})
 
+	if errors.Is(err, ErrMaxRemoveTxnRetries) {
```

Comment:
Is there any way for this to be an other error message? If there is an error, the query is retried, no?

Also, even if there would be another error - we don't care. This is the cleanup phase that runs on a best effort principle. Any error is just acknowledged/logged and we keep on going.

Reply:
We catch the error here to make sure if we encountered an error in future changes, we're not going to log the wrong message. I agree with the best effort and will make related changes.

Reply:
I would just move the log from line 62 to line 54 (before returning due to max tries) and then just 
```
if err != nil {
		log.Error(err, "Failed to wait for node to finish microcluster join before removing. Continuing with the cleanup...")
	}